rama-http-types/src/curl/traits.rs

```diff
@@ -0,0 +1,55 @@
+use crate::{Body, Request, Parts}
+use crate::dep::http_body;
+use bytes::Bytes;
+use rama_core::error::{BoxError, OpaqueError};
+
+mod sealed {
+    pub trait Sealed {}
```

Comment:
Good start, but better make 2 sealed traits, and call them the same as the public trait but with the `Priv` suffix. Also move the actual functions into the private trait, this way you also avoid that others can make use of the methods, giving us the flexibility to still change this.

So it would become something like:

```
trait IntoCurlHeadersCommandPriv {
    fn into_curl_headers_command(&self) -> Result<String, OpaqueError>;
}

pub trait IntoCurlHeadersCommand: IntoCurlHeadersCommandPriv {}
```